contracts/multisig-prover/src/execute.rs

```diff
@@ -317,8 +310,28 @@ pub fn should_update_worker_set(
     cur_workers: &WorkerSet,
     max_diff: usize,
 ) -> bool {
-    new_workers.signers.difference(&cur_workers.signers).count()
-        + cur_workers.signers.difference(&new_workers.signers).count()
+    let new_workers_signers = new_workers
+        .signers
+        .iter()
+        .map(|(address, signer)| Signer {
+            address: Addr::unchecked(address.clone()),
+            weight: signer.clone().0,
+            pub_key: signer.clone().1,
+        })
+        .collect::<BTreeSet<Signer>>();
+
+    let cur_workers_signers = cur_workers
+        .signers
+        .iter()
+        .map(|(address, signer)| Signer {
+            address: Addr::unchecked(address.clone()),
+            weight: signer.clone().0,
+            pub_key: signer.clone().1,
+        })
+        .collect::<BTreeSet<Signer>>();
```

Comment:
Hmm, this is a bit clunky now that signers is not a BTreeSet. I don't know how I feel about creating a BTreeSet just for this. I have a feeling it would be faster with just vectors, especially since the inputs are likely to be small. Can you try it with just vectors (i.e. convert signers to a vector, and then just compare the vectors), and see which is faster, vector comparison versus BTreeSet?

Reply:
Using a Vector is much faster, I'll update the PR to use them instead. Below are the results and the code used to test.

> Time elapsed in BTreeSet is: 51.016µs
> Time elapsed in Vector is: 13.716µs
> test execute::tests::should_update_worker_set_no_change ... ok

> Time elapsed in BTreeSet is: 41.678µs
> Time elapsed in Vector is: 14.988µs
> test execute::tests::should_update_worker_set_one_more ... ok

> Time elapsed in BTreeSet is: 40.916µs
> Time elapsed in Vector is: 14.437µs
> test execute::tests::should_update_worker_set_one_less ... ok

> Time elapsed in BTreeSet is: 40.737µs
> Time elapsed in Vector is: 13.055µs
> test execute::tests::should_update_worker_set_one_more_higher_threshold ... ok

> Time elapsed in BTreeSet is: 33.583µs
> Time elapsed in Vector is: 13.856µs
> test execute::tests::should_update_worker_set_diff_pub_key ... ok

Code used for testing:
```
pub fn should_update_worker_set(
    new_workers: &WorkerSet,
    cur_workers: &WorkerSet,
    max_diff: usize,
) -> bool {
    let mut result = false;
    {
        let start = Instant::now();
        
        let new_workers_signers = new_workers
            .signers
            .iter()
            .map(|(address, (weight, pub_key))| Signer {
                address: Addr::unchecked(address.clone()),
                weight: weight.clone(),
                pub_key: pub_key.clone(),
            })
            .collect::<BTreeSet<Signer>>();

        let cur_workers_signers = cur_workers
            .signers
            .iter()
            .map(|(address, (weight, pub_key))| Signer {
                address: Addr::unchecked(address.clone()),
                weight: weight.clone(),
                pub_key: pub_key.clone(),
            })
            .collect::<BTreeSet<Signer>>();

        result = new_workers_signers.difference(&cur_workers_signers).count()
            + cur_workers_signers.difference(&new_workers_signers).count()
            > max_diff;

        let duration = start.elapsed();
        println!("Time elapsed in BTreeSet is: {:?}", duration);
    }

    {
        let start = Instant::now();
        
        let new_workers_signers = new_workers
            .signers
            .iter()
            .map(|(address, (weight, pub_key))| Signer {
                address: Addr::unchecked(address.clone()),
                weight: weight.clone(),
                pub_key: pub_key.clone(),
            })
            .collect::<Vec<Signer>>();

        let cur_workers_signers = cur_workers
            .signers
            .iter()
            .map(|(address, (weight, pub_key))| Signer {
                address: Addr::unchecked(address.clone()),
                weight: weight.clone(),
                pub_key: pub_key.clone(),
            })
            .collect::<Vec<Signer>>();



        result = new_workers_signers.iter().filter(|item| !cur_workers_signers.contains(item)).count()
            + cur_workers_signers.iter().filter(|item| !new_workers_signers.contains(item)).count()
            > max_diff;

        let duration = start.elapsed();
        println!("Time elapsed in Vector is: {:?}", duration);
    }

    result
}
```

Reply:
yeah I figured